lib/bolt_server/transport_app.rb

```diff
@@ -135,11 +135,30 @@ def task_helper(target, task, parameters)
         end
       end
 
-      @executor.run_task(target, task, parameters).each do |result|
-        value = result.value
-        next unless value.is_a?(Hash)
-        next unless value.key?('_sensitive')
-        value['_sensitive'] = value['_sensitive'].unwrap
+      if timeout
+        task_thread = Thread.new do
+          @executor.run_task(target, task, parameters).each do |result|
+            value = result.value
+            next unless value.is_a?(Hash)
+            next unless value.key?('_sensitive')
+            value['_sensitive'] = value['_sensitive'].unwrap
+          end
+        end
+        # Wait for the timeout for the task to execute in the thread. If `join` times out, result will be nil.
+        if task_thread.join(timeout).nil?
+          task_thread.kill
+          raise Bolt::Error.new("Task execution on #{target.first.safe_name} timed out after #{timeout} seconds",
+                                'boltserver/task-timeout')
```

Comment:
```suggestion
                                'bolt-server/task-timeout')
```

Reply:
I copied from https://github.com/puppetlabs/bolt/blob/a45eb27a775da0d9f0d2216c7d2c5477403bef84/lib/bolt_server/transport_app.rb#L874 and https://github.com/puppetlabs/bolt/blob/a45eb27a775da0d9f0d2216c7d2c5477403bef84/lib/bolt_server/transport_app.rb#L853